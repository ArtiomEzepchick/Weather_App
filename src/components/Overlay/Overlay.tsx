import React from "react"
import classNames from "classnames"

import "./index.scss"

type Props = {
    isModalOpen: boolean;
    children?: JSX.Element;
}

const Overlay: React.FC<Props> = ({ isModalOpen, children }) => {
    return (
        <div
            className={classNames("overlay", isModalOpen && "show")}
            data-testid='overlay'
        >
            {children}
        </div>
    )
}

export default Overlay