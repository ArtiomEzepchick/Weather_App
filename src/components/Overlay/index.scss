.overlay {
  display: none;
  position: fixed;
  opacity: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 100;

  &.show {
    visibility: visible;
    display: block;
    opacity: 1;
    overflow: hidden;
    background-color: #7c7c7cf0;
    animation: fadeInFromNone 0.3s ease-in-out;
    animation-fill-mode: forwards;
  }

  &.hidden {
    animation: fadeOutFromBlock 0.3s ease-in-out;
  }
}

@keyframes fadeInFromNone {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@keyframes fadeOutFromBlock {
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0;
  }
}
