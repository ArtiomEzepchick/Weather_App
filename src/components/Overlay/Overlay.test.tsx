import React from "react"
import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom'

import Overlay from "./Overlay"

describe('Overlay', () => {
    it("renders with 'show' class when isModalOpen equals true", () => {
        render(<Overlay isModalOpen={true} />)

        const overlay = screen.getByTestId<HTMLDivElement>('overlay')

        expect(overlay).toBeInTheDocument()
        expect(overlay).toHaveClass('show')
    })

    it("renders with 'show' class when isModalOpen equals true and with children", () => {
        const mockChild: JSX.Element = <p>Content</p>

        const { getByText } = render(<Overlay isModalOpen={true} children={mockChild}/>)

        const overlay = screen.getByTestId<HTMLDivElement>('overlay')

        expect(overlay).toBeInTheDocument()
        expect(overlay).toHaveClass('show')
        expect(getByText(/Content/i)).toBeInTheDocument()
    })

    it("renders without 'show' class when isModalOpen equals false", () => {
        render(<Overlay isModalOpen={false} />)

        const overlay = screen.getByTestId<HTMLDivElement>('overlay')
        
        expect(overlay).toBeInTheDocument()
        expect(overlay).not.toHaveClass('show')
    })
})