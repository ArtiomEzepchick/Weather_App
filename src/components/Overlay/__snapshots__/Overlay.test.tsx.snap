// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay renders with 'show' class when isModalOpen equals true 1`] = `
<div
  class="overlay show"
  data-testid="overlay"
/>
`;

exports[`Overlay renders with 'show' class when isModalOpen equals true and with children 1`] = `
<div
  class="overlay show"
  data-testid="overlay"
>
  <p>
    Content
  </p>
</div>
`;

exports[`Overlay renders without 'show' class when isModalOpen equals false 1`] = `
<div
  class="overlay"
  data-testid="overlay"
/>
`;
